src/java/com/google/cloud/bigquery/dwhassessment/hooks/MigrationAssessmentLoggingHook.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigquery.dwhassessment.hooks;
+
+import static org.apache.hadoop.hive.ql.hooks.Entity.Type.PARTITION;
+import static org.apache.hadoop.hive.ql.hooks.Entity.Type.TABLE;
+
+import com.google.cloud.bigquery.dwhassessment.hooks.avro.AvroSchemaLoader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.generic.GenericRecordBuilder;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.ql.QueryPlan;
+import org.apache.hadoop.hive.ql.exec.Utilities;
+import org.apache.hadoop.hive.ql.exec.mr.ExecDriver;
+import org.apache.hadoop.hive.ql.exec.tez.TezTask;
+import org.apache.hadoop.hive.ql.hooks.Entity;
+import org.apache.hadoop.hive.ql.hooks.ExecuteWithHookContext;
+import org.apache.hadoop.hive.ql.hooks.HookContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Hook, which captures query events and stores them as {@link GenericRecord}. */
+public class MigrationAssessmentLoggingHook implements ExecuteWithHookContext {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(MigrationAssessmentLoggingHook.class.getName());
+
+  private static final Schema QUERY_EVENT_SCHEMA = AvroSchemaLoader.loadSchema("QueryEvents.avsc");
+
+  // Just for the example. Real hook will write it to the file
+  public ArrayList<GenericRecord> records = new ArrayList<>();
+
+  public void run(HookContext hookContext) throws Exception {
+    processQueryEventAvro(hookContext);
+  }
+
+  private void processQueryEventAvro(HookContext hookContext) {
+    QueryPlan plan = hookContext.getQueryPlan();
+
+    LOG.info("Received hook notification for: " + plan.getQueryId());
```

Comment:
The logger supports parameter placeholders, so that the concatenation can be performed only after the logging event is enabled for logging:

```
LOG.info("Received hook notification for: '{}'", plan.getQueryId());
```

Similarly below.

Reply:
Done